ARCHITECTURE.md

```diff
@@ -0,0 +1,116 @@
+
+The Git for Windows project is also responsible for keeping the Windows part of `git/git`'s automated builds up and running. On Windows, there is no canonical and easy way to get a build environment necessary to build Git and run its test suite, therefore this is a non-trivial task that comes with its own maintenance cost. Git for Windows provides two GitHub Actions to help with that: [`git-for-windows/setup-git-for-windows-sdk`](https://github.com/git-for-windows/setup-git-for-windows-sdk) to set up a tiny subset of Git for Windows' full SDK (which would require about 500MB to be cloned, as opposed to the ~75MB of that subset) and [`git-for-windows/get-azure-pipelines-artifact`](https://github.com/git-for-windows/get-azure-pipelines-artifact) e.g. to download some regularly pre-built artifacts (for example, when `git/git`'s automated tests ran on an Ubuntu version that did not provide an up to date [Coccinelle](https://coccinelle.gitlabpages.inria.fr/website/) package, this GitHub Action was used to download a pre-built version of that Debian package).
+
+## Maintaining Git for Windows' components
+
+Git for Windows uses a combination of [a GitHub App called GitForWindowsHelper](https://github.com/git-for-windows/gfw-helper-github-app) (to listen for so-called [slash commands](https://github.com/git-for-windows/gfw-helper-github-app#slash-commands)) combined with workflows in [the `git-for-windows-automation` repository](https://github.com/git-for-windows/git-for-windows-automation/) (for computationally heavy tasks) to support Git for Windows' repetitive tasks.
+
+This heavy automation serves two purposes:
+
+1. Document the knowledge about "how things are done" in the Git for Windows project.
+2. Make Git for Windows' maintenance less tedious by off-loading as many tasks onto machines as possible.
+
+One neat trick of some `git-for-windows-automation` workflows is that they "mirror back" check runs to the targeted PRs in another repository. This essentially allows versioning the source code independently of the workflow definition.
+
+Here is a diagram showing how the bits and pieces fit together.
+
+```mermaid
+graph LR
+  A[`monitor-components`] --> |opens| B
+  B{issues labeled<br />`component-update`} --> |/open pr| C
+  C((GitForWindowsHelper)) --> |triggers| D
+  D[`open-pr`] --> |opens| E
+  E{PR in</br>MINGW-packages<br />MSYS2-packages<br />build-extra} --> |closes| B
+  E --> |/deploy| F
+  F((GitForWindowsHelper)) --> |triggers| G
+  G[`build-and-deploy`] --> |deploys to| H
+  H{Pacman repository}
+  C --> |backed by| I
+  F --> |backed by| I
+  I[[Azure Function]]
+  D --> |running in| J
+  G --> | running in| J
+  J[[git-for-windows-automation]]
+  K[[git-sdk-32<br />git-sdk-64<br />git-sdk-arm64]] --> |syncing from| H
+  B --> |/add release note| L
+  L[`add-release-note`]
+```
```

Comment:
This diagram is great! 🧜‍♀️ It makes this whole setup (and the remainder of your documentation here) much easier to follow.